OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -87,6 +87,16 @@ void Frame::addMeshGeometry(const std::string& aGeometryFileName, const SimTK::V
     addGeometry(geom);
 }
 
+
+void Frame::attachGeometry(const OpenSim::Geometry& geom, const SimTK::Vec3 scale)
+{
+    SimTK::ClonePtr<Geometry> clone = SimTK::ClonePtr<Geometry>(geom);
+    clone->set_scale_factors(scale);
+    clone->setFrameName(getName());
+    addGeometry(clone.updRef());
```

Comment:
Actually, there might be two copies: one by the constructor, and one by the assignment. The compiler might optimize away the copy assignment, though I'm not sure.